### PR TITLE
Corrige los fallos cuando las pruebas se ejecutan en máquinas que usan castellano como lenguaje.

### LINKED ISSUE #1967: Run build and tests on Spanish image
Consider running SignalR build and tests on a Spanish image to catch culture related bugs.

Roslyn already does this - https://github.com/aspnet/SignalR/issues/1905#issuecomment-380648062

// @jaredpar @pranavkm

### PR DESCRIPTION
**Pull request by anurse**
... I hope Google Translate didn't make me look too stupid there ;).

This should fix the issues with the Spanish build. Just a few places where we hard-coded asserts of things that end up getting localized.

Fixes #1967

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.ConnectionLifecycle.cs [hunk 1]
```diff
@@ -509,7 +509,10 @@ Task<ConnectionContext> ConnectionFactory(TransferFormat format)
                 {
                     var startTask = hubConnection.StartAsync(cts.Token);
                     var exception = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => startTask.OrTimeout());
-                    Assert.Equal("The operation was canceled.", exception.Message);
+
+                    // REVIEW: Alternatively, we could just not assert the message...
```
Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.ConnectionLifecycle.cs]: Thoughts on this? If we don't care about the exact message, we should just not assert it. If we care that it's the "Default" message, we should use the constructor to get the actual string so it's localized.

Comment (mikaelm12) [reply test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.ConnectionLifecycle.cs]: Yeah, I think in this case the exact message is not important. This seems like a fine fix to me

File: test/Microsoft.AspNetCore.SignalR.Tests.Utils/ExceptionMessageExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using System;
```
Comment (mikaelm12) [inline test/Microsoft.AspNetCore.SignalR.Tests.Utils/ExceptionMessageExtensions.cs]: Header  😄

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.ConnectionLifecycle.cs (hunk 1 above)
Comment (BrennanConroy) [reply test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.ConnectionLifecycle.cs]: I think we don't need this message

File: test/Microsoft.AspNetCore.SignalR.Tests.Utils/ExceptionMessageExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Microsoft.AspNetCore.SignalR.Tests
+{
+    public static class ExceptionMessageExtensions
+    {
+        public static string GetLocalizationSafeMessage(this ArgumentException argex)
```
Comment (mikaelm12) [inline test/Microsoft.AspNetCore.SignalR.Tests.Utils/ExceptionMessageExtensions.cs]: nit camel case `argex` to `argEx`

**Comment by anurse** [comment]
I'm probably going to change the commit message when I squash it to remove the terrible spanish btw ;)

**Comment by javiercn** [comment]
@anurse @davidfowl I fixed the PR title for you :)

**Comment by anurse** [comment]
Filed a bug for the flaky test

**Pull request merged**